### ISSUE TITLE
Task terms not validating correctly when activating

### ISSUE THREAD

**Issue by ajefts** [appirio-tech/direct-app#192]
When you save a task as a draft and have added a registrant, it is successfully validating if the user has agreed to the proper terms for the challenge.

However, if  you fill out the entire challenge and activate without saving as a draft, it silently does not add the registrant.

**Comment by skyhit** [comment]
@ajefts I was thinking about this, I think we should validate just when we enter the handles in the input box, instead of when saving or activating.  which mean it should be a separate logic.

**Comment by ajefts** [comment]
@skyhit I agree.  That would be better for sure.  What would trigger the validation?  Should we add a user picker at the same time?

**Comment by skyhit** [comment]
@ajefts for terms, it is actually set for project level, not challenge level, right?

I would consider to have a user picker similar as skills in profiles, so a search suggestion for a list of members, and when selected, it will validate if user has terms and added to the select users.

I wondering if we can define some reusable component, which may fit for many cases.

**Comment by ajefts** [comment]
Terms are set on the challenge level (common_oltp.project_role_terms_of_use_xref).  We're able to configure a billing account with terms so that every time a challenge is created with Billing Account A, it'll apply the terms to the challenge.

For the user picker, yes...I think it would be good to have a component we can reuse.  We have current user pickers to add copilots and users to projects, but it's not very good.
